### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by paulcurren**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/app.py [hunk 1]
```diff
@@ -10,12 +12,29 @@
...
 
 
 @app.route('/add_item', methods=['POST'])
 def addItem():
     title = request.form.get('title')
     add_item(title)
-    return render_template('index.html', items = get_items())
+    statuses = get_statuses()
+    allItems = get_items()
+    return render_template('index.html', statuses = list(statuses), items = list(allItems))
```
Comment (BenRamchandani) [inline todo_app/app.py]: Minor, but redirecting in response to a successful POST form submission plays slightly nicer with browser behaviour (e.g. refreshing), and means you don't need to duplicate code.

```suggestion
    return redirect('/', 303)
```

File: todo_app/data/trello_items.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    __add_card(listId, title)
+
+
+
+def save_item(item) -> Item:
+    return item
+
+
+def __add_card(listId, name):
+    response = requests.post(f"https://api.trello.com/1/cards?key={__key}&token={__token}&name={name}&idList={listId}")
```
Comment (BenRamchandani) [inline todo_app/data/trello_items.py]: It's generally better to let the library build the query params, for the same reason you wouldn't concatenate strings into an SQL query, e.g. what if name is `Some_title&desc=Injected_description`? `requests.post` has a `params` keyword argument for this.

File: todo_app/templates/index.html [hunk 3]
```diff
@@ -6,23 +6,48 @@
     <h1 class="display-4">To-Do App</h1>
     <p class="lead">Just another to-do app.</p>
   </div>
+  <iframe name="hiddenFrame" width="0" height="0" border="0" style="display: none;"></iframe>
```
Comment (BenRamchandani) [inline todo_app/templates/index.html]: What's this for?

File: todo_app/app.py [hunk 4]
```diff
@@ -1,7 +1,9 @@
 from flask import Flask, render_template, request
 
 from todo_app.flask_config import Config
-from todo_app.data.session_items import add_item, get_items
+#from todo_app.data.session_items import add_item, get_items
```
Comment (BenRamchandani) [inline todo_app/app.py]: You can delete this and session_items.py now

**Review by BenRamchandani — COMMENTED** [review]
Looks good, code is neat and it all works for me.

I'll approve this on Aptem, but before you merge please update your `.env.template` file and README with the setup steps required to get the app running and connected to Trello.

I've left some other optional comments below.

File: todo_app/app.py (hunk 4 above)
Comment (paulcurren) [reply todo_app/app.py]: done

File: todo_app/templates/index.html (hunk 3 above)
Comment (paulcurren) [reply todo_app/templates/index.html]: removed

File: todo_app/data/trello_items.py (hunk 2 above)
Comment (paulcurren) [reply todo_app/data/trello_items.py]: done

File: todo_app/app.py (hunk 1 above)
Comment (paulcurren) [reply todo_app/app.py]: done

**Pull request merged**